lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -112,10 +121,32 @@ def self.available_options
           FastlaneCore::ConfigItem.new(
             key: :skip_if_exists,
             description: 'If the file already exists in the S3 bucket, skip the upload (and report it in the logs), instead of failing with `user_error!`',
+            deprecated: 'Use if_exists instead',
+            conflicting_options: [:if_exists],
+            conflict_block: proc do |option|
+              UI.user_error!("You cannot set both :#{option.key} and :skip_if_exists. Please only use :if_exists.")
+            end,
             optional: true,
-            default_value: false,
             type: Boolean
           ),
+          FastlaneCore::ConfigItem.new(
+            key: :if_exists,
+            description: 'What do to if the file file already exists in the S3 bucket. Possible values :skip, :replace, :fail. When set, overrides the deprecated skip_if_exists option',
+            conflicting_options: [:skip_if_exists],
+            conflict_block: proc do |option|
+              UI.user_error!("You cannot set both :#{option.key} and :if_exists. Please only use :if_exists.")
+            end,
+            optional: true,
+            type: Symbol,
+            # We cannot set a default value and have backward compatibility with skip_if_exists at the same time.
+            # (Short of duplicating the default value knowledge in the action implementation)
+            #
+            # We have a test for the default behavior which should hopefully remind us to uncomment this line once well remove skip_if_exists.
+            # default_value: :fail,
+            verify_block: proc do |value|
+              UI.user_error!('`if_exist` must be one of :skip, :replace, :fail') unless %i[skip replace fail].include?(value)
+            end
```

Comment:
The suggestion from https://github.com/wordpress-mobile/release-toolkit/pull/495#discussion_r1235606551 was to set `default_value: :fail`.

I like that but didn't realize it would break the scenario where this is not explicitly set and we want `skif_if_exists` to take over.

I think there could have been a way to set this and move all the fallback logic in the implementation when keeping in mind that we don't allow both options to be set at the same time, but the resulting code would have been more complex. I think that between this comment and the unit tests that assert the default behavior we should be covered once it's time to remove `skip_if_exists`.

Reply:
Ah yeah good point. In fact, that reminds me of [something similar I saw in `Supply::Options#available_options` too](https://github.com/fastlane/fastlane/blob/0495da3504f8506587161c5ffb468f9694e65635/supply/lib/supply/options.rb#L96), So I think it makes sense to have it that way until we remove legacy.